packages/babel-plugin/src/css-prop/__tests__/object-literal.test.tsx

```diff
@@ -371,8 +371,50 @@ describe('css prop object literal', () => {
         <div css={{ color: 'blue', ':hover': mixin() }}>hello world</div>
       `);
 
-    expect(actual).toInclude(`.css-test{color:blue}`);
-    expect(actual).toInclude('.css-test:hover{color:red}');
+    expect(actual).toInclude(`.cc-hash-test{color:blue}`);
+    expect(actual).toInclude('.cc-hash-test:hover{color:red}');
+  });
+
+  it('should transform complex object with no argument functions', () => {
+    const actual = transform(`
+        import '@compiled/core';
+        import React from 'react';
+
+        const bgColor = 'blue';
+        const fontSize = 12;
+        const fontStyling = {
+          weight: 500,
+          style: 'italic',
+          family: 'sans-serif',
+        };
+
+        const mixin1 = () => ({ color: 'red', backgroundColor: bgColor });
+        const mixin2 = function() { return { fontStyle: fontStyling.style } };
+        function mixin3() { return { fontFamily: fontStyling.family } };
+
+        const sizes = {
+          mixin1: () => \`1px solid \${bgColor}\`,
+          mixin2: () => ({ fontSize }),
+          mixin3: function() {return {fontWeight: fontStyling.weight};}
+        };
+
+        <div css={{
+          color: 'blue',
+          ':hover': mixin1(),
+          border: sizes.mixin1(),
+          ...sizes.mixin2(),
+          ...sizes.mixin3(),
+          ...mixin2(),
+          ...mixin3(),
+        }}>
+          hello world
+        </div>
+      `);
+
+    expect(actual).toInclude(
```

Comment:
i'm impressed!

thoughts about breaking this test down so its not as intimidating? 😅